pkgs/by-name/am/amazing-marvin/package.nix

```diff
@@ -0,0 +1,37 @@
+{
+  appimageTools,
+  fetchurl,
+  lib,
+}:
+
+let
+  pname = "amazing-marvin";
+  version = "1.66.2";
+
+  src = fetchurl {
+    url = "https://amazingmarvin.s3.amazonaws.com/Marvin-${version}.AppImage";
+    sha256 = "sha256-T3tkLLZPLhb6NdOwr2xBVJAqDOPvkFhzxPIJJEQ7cyY=";
+  };
+
+  appimageContents = appimageTools.extract { inherit pname version src; };
+
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+  extraInstallCommands = ''
+    install -m 444 -D ${appimageContents}/marvin.desktop $out/share/applications/marvin.desktop
+    install -m 444 -D ${appimageContents}/usr/share/icons/hicolor/512x512/apps/marvin.png \
+      $out/share/icons/hicolor/512x512/apps/marvin.png
+    substituteInPlace $out/share/applications/marvin.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
```

Comment:
```suggestion
      --replace-fail 'Exec=AppRun' 'Exec=${pname}'
```
- #260776

Reply:
And use either meta.mainProgram or `amazing-marvin` #277994